### PR TITLE
Add some info to `ThreadPool.SetMinThreads` on how increasing the minimum threads may degrade performance

### PR DESCRIPTION
**Pull request by kouvel**

### CODE REVIEW

File: xml/System.Threading/ThreadPool.xml [hunk 1]
```diff
@@ -1265,8 +1265,15 @@ If a thread pool implementation may have different types of work items, the coun
...
+ - Context switching between threads may increase CPU usage.
+ - Garbage collection may take more CPU time in thread stack walking.
+ - The process may consume more memory.
+  
 > [!CAUTION]
->  By default, the minimum number of threads is set to the number of processors on a system. You can use the <xref:System.Threading.ThreadPool.SetMinThreads%2A> method to increase the minimum number of threads. However, unnecessarily increasing these values can cause performance problems. If too many tasks start at the same time, all of them might appear to be slow. In most cases, the thread pool will perform better with its own algorithm for allocating threads. Reducing the minimum to less than the number of processors can also hurt performance.  
+>  Using the <xref:System.Threading.ThreadPool.SetMinThreads%2A> method to increase the minimum number of threads can cause performance problems as described above. In most cases, the thread pool will perform better with its own algorithm for allocating threads. Reducing the minimum to less than the number of processors can also hurt performance.
```
Comment (gewarren) [inline xml/System.Threading/ThreadPool.xml:1276]:
```suggestion
> Using the <xref:System.Threading.ThreadPool.SetMinThreads%2A> method to increase the minimum number of threads can cause performance problems as described in the preceding text. In most cases, the thread pool will perform better with its own algorithm for allocating threads. Reducing the minimum to less than the number of processors can also hurt performance.  
```

File: xml/System.Threading/ThreadPool.xml [hunk 2]
```diff
@@ -1265,8 +1265,15 @@ If a thread pool implementation may have different types of work items, the coun
   
  If you specify a negative number or a number larger than the maximum number of active thread pool threads (obtained using <xref:System.Threading.ThreadPool.GetMaxThreads%2A>), <xref:System.Threading.ThreadPool.SetMinThreads%2A> returns `false` and does not change either of the minimum values.  
   
+ By default, the minimum number of threads is set to the processor count. <xref:System.Threading.ThreadPool.SetMinThreads%2A> may be used to increase the minimum number of threads, such as to temporarily work around issues where some queued work items or tasks block thread pool threads, which sometimes may lead to a situation where all worker or I/O completion threads are blocked (starvation). However, increasing the minimum number of threads may degrade performance in other ways, such as:
```
Comment (gewarren) [inline xml/System.Threading/ThreadPool.xml:1268]:
```suggestion
 By default, the minimum number of threads is set to the processor count. You can use <xref:System.Threading.ThreadPool.SetMinThreads%2A> to increase the minimum number of threads, such as to temporarily work around issues where some queued work items or tasks block thread pool threads. Those blockages sometimes lead to a situation where all worker or I/O completion threads are blocked (starvation). However, increasing the minimum number of threads might degrade performance in other ways, such as:
 
```

File: xml/System.Threading/ThreadPool.xml [hunk 3]
```diff
@@ -1265,8 +1265,15 @@ If a thread pool implementation may have different types of work items, the coun
...
+ - The thread pool may schedule more worker threads, even when the worker threads are not getting blocked. The oversubscription can cause threads that get scheduled-out to be significantly delayed as they wait in a long queue to get another time slice, which may lead to some work items or tasks being delayed.
```
Comment (gewarren) [inline xml/System.Threading/ThreadPool.xml:1269]:
```suggestion
 - The thread pool may schedule more worker threads, even when the worker threads are not getting blocked. The oversubscription can cause threads that get scheduled-out to be significantly delayed as they wait in a long queue to get another time slice, delaying some work items or tasks.
```

File: xml/System.Threading/ThreadPool.xml [hunk 4]
```diff
@@ -1265,8 +1265,15 @@ If a thread pool implementation may have different types of work items, the coun
...
+ - Worker threads may take more CPU time in dequeuing work items due to there being more threads to scan for work-stealing.
```
Comment (gewarren) [inline xml/System.Threading/ThreadPool.xml:1270]:
```suggestion
 - Worker threads may take more CPU time in dequeuing work items due to having to scan more threads to steal work from.
```

**Review by gewarren — APPROVED** [review]
I left a few suggestions that should help with localization etc.

File: xml/System.Threading/ThreadPool.xml (hunk 1 above)
Comment (kouvel) [reply xml/System.Threading/ThreadPool.xml]: Nice catch, thanks!

**Pull request merged**